persistence/jackson/src/test/java/ai/timefold/solver/jackson/api/TimefoldJacksonModuleTest.java

```diff
@@ -41,6 +57,126 @@ void polymorphicScore() {
+
+        var serialized = objectMapper.writeValueAsString(originalScoreAnalysis);
+        Assertions.assertThat(serialized)
+                .isEqualToIgnoringWhitespace("""
+                        {
+                            "score" : "2hard/4soft",
+                            "constraints" : [ {
+                              "id" : "package1/constraint1",
+                              "score" : "2hard/0soft",
+                              "matches" : [ {
+                                "score" : "1hard/0soft",
+                                "justification" : [ "A", "B" ]
+                              }, {
+                                "score" : "1hard/0soft",
+                                "justification" : [ "B", "C", "D" ]
+                              } ]
+                            }, {
+                              "id" : "package2/constraint2",
+                              "score" : "0hard/4soft",
+                              "matches" : [ {
+                                "score" : "0hard/1soft",
+                                "justification" : [ "D" ]
+                              }, {
+                                "score" : "0hard/3soft",
+                                "justification" : [ "A", "C" ]
+                              } ]
+                            } ]
+                          }""");
+
+        objectMapper.registerModule(new CustomJacksonModule());
+        ScoreAnalysis<HardSoftScore> deserialized = objectMapper.readValue(serialized, ScoreAnalysis.class);
+        Assertions.assertThat(deserialized).isEqualTo(originalScoreAnalysis);
+    }
+
+    public static final class CustomJacksonModule extends SimpleModule {
+
+        public CustomJacksonModule() {
+            super("Timefold Custom");
+            addDeserializer(ScoreAnalysis.class, new CustomScoreAnalysisJacksonDeserializer());
+        }
+
+    }
+
+    public static final class CustomScoreAnalysisJacksonDeserializer
+            extends AbstractScoreAnalysisJacksonDeserializer<HardSoftScore> {
+
+        @Override
+        protected HardSoftScore parseScore(String scoreString) {
+            return HardSoftScore.parseScore(scoreString);
+        }
+
+        @Override
+        protected <ConstraintJustification_ extends ConstraintJustification> ConstraintJustification_
+                parseConstraintJustification(ConstraintRef constraintRef, String constraintJustificationString,
+                        HardSoftScore score) {
+            List<Object> justificationList = Arrays.stream(constraintJustificationString.split(","))
+                    .map(s -> s.replace("[", "")
+                            .replace("]", "")
+                            .replace("\"", "")
+                            .strip())
+                    .collect(Collectors.toList());
+            return (ConstraintJustification_) DefaultConstraintJustification.of(score, justificationList);
```

Comment:
Can the justification be just any object and it's a list just in this example, or is it always a list?

Reply:
In this case, it is always a list, because the justification is always `DefaultConstraintJustification`. In general, no, this can be anything.